### PR TITLE
CustomizableUI.jsm can't be used for Pale Moon - setting conditions

### LINKED ISSUE #97: DevTools - CustomizableUI can't be used for Pale Moon
Some preparation for `Pale Moon UXP`.

### PR DESCRIPTION
**Pull request by janekptacijarabaci**
This resolves #97 → You add the label `Component: Devtools`, please

Or is there any better solution for this?

---

I've created the new build (x32, Windows) - `Basilisk`/`Pale Moon UXP` - and tested.

### CODE REVIEW

File: devtools/client/framework/devtools-browser.js [hunk 1]
```diff
@@ -27,9 +27,12 @@ loader.lazyRequireGetter(this, "DebuggerServer", "devtools/server/main", true);
 loader.lazyRequireGetter(this, "DebuggerClient", "devtools/shared/client/main", true);
 loader.lazyRequireGetter(this, "BrowserMenus", "devtools/client/framework/browser-menus");
 
-loader.lazyImporter(this, "CustomizableUI", "resource:///modules/CustomizableUI.jsm");
 loader.lazyImporter(this, "AppConstants", "resource://gre/modules/AppConstants.jsm");
 
+if (AppConstants.MOZ_APP_NAME.toLowerCase() != "palemoon") {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: Please do NOT use AppConstants.. That will go away...

Do like I did in 8f6d3dab81c7f8f97ef197e26ab9439b09735b8f

You will need to locate the final target path for this file in dist and then take it out of the crappy mozbuild function magic and put the file into preprocessing..

You would then ifdef it with the conditional `MC_BASILISK`.

```
#ifdef MC_BASILISK
  loader.lazyImporter(this, "CustomizableUI", "resource:///modules/CustomizableUI.jsm");
#endif
```

File: devtools/client/framework/devtools-browser.js [hunk 2]
```diff
@@ -295,6 +298,9 @@ var gDevToolsBrowser = exports.gDevToolsBrowser = {
    * Install Developer widget
    */
   installDeveloperWidget: function () {
+    if (typeof CustomizableUI === "undefined") {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: In these conditionals...

#ifdef MC_BASILISK
  //all the code
#else 
  return whatever;
#endif

File: devtools/client/framework/devtools-browser.js [hunk 3]
```diff
@@ -350,7 +356,7 @@ var gDevToolsBrowser = exports.gDevToolsBrowser = {
    */
   // Used by itself
   installWebIDEWidget: function () {
-    if (this.isWebIDEWidgetInstalled()) {
+    if ((typeof CustomizableUI === "undefined") || this.isWebIDEWidgetInstalled()) {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: ditto

File: devtools/client/framework/devtools-browser.js [hunk 4]
```diff
@@ -374,6 +380,9 @@ var gDevToolsBrowser = exports.gDevToolsBrowser = {
   },
 
   isWebIDEWidgetInstalled: function () {
+    if (typeof CustomizableUI === "undefined") {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: ditto

File: devtools/client/framework/devtools-browser.js [hunk 5]
```diff
@@ -387,6 +396,9 @@ var gDevToolsBrowser = exports.gDevToolsBrowser = {
    * Uninstall WebIDE widget
    */
   uninstallWebIDEWidget: function () {
+    if (typeof CustomizableUI === "undefined") {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: ditto

File: devtools/client/framework/devtools-browser.js [hunk 6]
```diff
@@ -398,6 +410,9 @@ var gDevToolsBrowser = exports.gDevToolsBrowser = {
    */
    // Used by webide.js
   moveWebIDEWidgetInNavbar: function () {
+    if (typeof CustomizableUI === "undefined") {
```
Comment (mattatobin) [inline devtools/client/framework/devtools-browser.js]: ditto

**Comment by mattatobin** [comment]
Blocks #102

**Comment by janekptacijarabaci** [comment]
If this is used:

```
#ifndef MC_PALEMOON
  loader.lazyImporter(this, "CustomizableUI", "resource:///modules/CustomizableUI.jsm");
#endif
```

and

```
#ifdef MC_PALEMOON
  return;
#endif
```

...it is likely that other frontends will use it (`CustomizableUI`).

Is it better than the original one (`#ifdef MC_BASILISK`)?

**Comment by mattatobin** [comment]
Then you end up with ureachable code warnings being emited and there is no reason australis code should be there.. Preprocessing it out as i indicated is the prefered solution..

**Comment by wolfbeast** [comment]
CustomizableUI is Australis-specific, so only in use in Basilisk on the platform.

**Pull request merged**